File: Data Structures/Arrays/Left Rotation/leftRotate.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the rotLeft function below.
def rotLeft(a, d):
    copy = a
    newA = []

    for i in range(d-1,len(a)-1):
        newA.insert(0,copy[-1])
        copy.pop()
    
    for numLeft in copy:
        newA.insert(len(newA),numLeft)

    return newA

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    nd = input().split()

    n = int(nd[0])

    d = int(nd[1])

    a = list(map(int, input().rstrip().split()))

    result = rotLeft(a, d)

    fptr.write(' '.join(map(str, result)))
    fptr.write('\n')

    fptr.close()

File: Data Structures/Arrays/Left Rotation/leftRotate.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the rotLeft function below.
function rotLeft(a, d) {
    var count = 0;
    var copy = a;
    var newA = [];

    for (let i = d-1; i < a.length - 1; i++) {
        newA[count] = copy[i+1];
        count += 1;
    }

    for (let i = 0; i < d; i++) {
        newA[count] = copy[i];
        count += 1;
    }

    return newA;

}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const nd = readLine().split(' ');

    const n = parseInt(nd[0], 10);

    const d = parseInt(nd[1], 10);

    const a = readLine().split(' ').map(aTemp => parseInt(aTemp, 10));

    const result = rotLeft(a, d);

    ws.write(result.join(' ') + '\n');

    ws.end();
}
